src/components/TodoList.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+import Todo from "./Todo";
+
+class TodoList extends Component {
+  constructor() {
+    super();
+    this.state = {
+      todos: [],
+      newTodo: {
+        index: null,
+        text: '',
+        completed: false
+      }
+    };
+  }
+
+  handleClick = index => {
+    const { todos } = this.state;
+    const clicked = todos.filter(obj => obj.index === index);
+    clicked[0].completed = !clicked[0].completed;
+    const newTodos = [...todos.slice(0, index), ...todos.slice(index)];
```

Comment:
Very nice! But I think you meant to do `[...todos.slice(0, index - 1), clicked[0], ...todos.slice(index)]`